docs: added docs for migrating encrypted secrets (#30132)
renovatebot/renovate: docs/usage/getting-started/migrating-secrets.md

# Migrating Secrets from Repo Config to App Settings

> Storing of encrypted secrets in the repo is not supported starting from 01-Sep-2024 for Mend Renovate Cloud apps.

Reviewer: I'm not sure this date is realistic if we haven't yet begun to warn users

Author: I'll bump it to 1-Oct-2024

Previously, you could encrypt a secret with the [Renovate encryption tool](https://app.renovatebot.com/encrypt) and then put it in your Renovate config file.
Going forward, all secrets must be stored in the App settings on the cloud.
They can be referenced from the Renovate config files inside the repo using `{{ secrets.SECRET_NAME }}` notation.

> 

Reviewer 2: ```suggestion
```

```json title "Deprecated: example of encrypted secrets in Renovate config"
{
  "hostRules": [
    {
      "matchHost": "github.com",
      "encrypted": {
        "token": "drsMDVf6M2hTZCN......+gQm/0Rpw"
      }
    }
  ]
}
```

> 

Reviewer 2: ```suggestion
```

**Example - Supported version**: Reference App secrets from the Renovate config

```json
{
  "hostRules": [
    {
      "matchHost": "github.com",
      "token": "{{ secrets.GITHUB_COM_TOKEN }}"
    }
  ]
}
```

> [!NOTE]
>
> When migrating encrypted secrets, we recommend copying over the existing secrets in their _**encrypted form**_. The UI will handle the decryption and store the value securely.
> Avoid trying to change the secret during migration, as this will introduce an additional potential point of failure.

## To migrate encrypted secrets using the encrypted value

1. Copy the encrypted secret from the Renovate config file.

2. Go to the settings for the correct Org or Repo in the web UI at [http://developer.mend.io](http://developer.mend.io).

3. In the **Credentials** page, press `ADD SECRET` to add the encrypted secret.

4. Provide a value for `Secret name`, put the encrypted secret in the `Secret Value` field, and press `SAVE`.

   > [!NOTE]
   >
   > When migrating an existing secret from a repo, be sure to add the secret to the same Org or Repo for which the secret was generated.
   >

>    > - A secret generated for a specific repo cannot be added to a different repo.

Reviewer 2: ```suggestion
   > - A secret generated for a specific repository can only be added to that _same_ repository.
```

>    >   - A secret generated for a specific repo cannot be added to the Org settings. It can only be added into Repo settings for the matching repo.

Reviewer 2: ```suggestion
   >   - A secret generated for a specific repository can only be added to the repository settings for the matching repository. This secret can _not_ be added to the organization's settings.
```

>    >   - A secret generated without a specific repo can be added into Org or Repo settings in the matching org.

Reviewer 2: ```suggestion
   >   - A secret generated without a specific repository _can_ be added into the organization _or_ into the repository settings under that organization.
```

>    The settings UI will provide helpful information if you try to add the secret in the wrong place.

Reviewer 2: ```suggestion
   When you try to add a secret in the wrong place, the settings UI will help you:
```

   ![Migrating secrets error](../assets/images/app-settings/encrypted-secrets-error.png)

> 5. Look for the confirmation box to say **“Successfully migrated secret”** to confirm that the secret was correctly decrypted and stored in the correct Org or Repo.

Reviewer 2: ```suggestion
5. When you see the confirmation box with the text: **“Successfully migrated secret”**, your secret is now stored in the correct organization or repository.
```

Do we need the detail about the decrypting? If yes, we should probably mention if/that you re-encrypted the value safely... :wink:

![Successfully migrated secret](../assets/images/app-settings/stored-secret-encrypted.png)

> [!IMPORTANT]
>

> > If the confirmation box does not say **“Successfully migrated secret”**, and instead says **“Successfully stored secret”** then the secret was not detected as a Renovate encrypted secret and the value was treated as plain text.

Reviewer 2: ```suggestion
> If the confirmation box does _not_ say **“Successfully migrated secret”**, but says: **“Successfully stored secret”** then the secret was not detected as a Renovate encrypted secret, and the value was treated as plaintext.
```

> If you were expecting to import a secret originally encrypted by Renovate:
>

> > - Make sure you are using the secret in the correct Org and/or Repo.

Reviewer 2: ```suggestion
> - Make sure you are pasting the secret into the correct organization or repository
```

> - Check that you copied the encrypted secret correctly, and try again.

## To migrate encrypted secrets using the plain text value

> 1. Go to the settings for the correct Org or Repo in the web UI at [http://developer.mend.io](http://developer.mend.io).

Reviewer 2: ```suggestion
1. Go to the settings for the correct organization or repository in the web UI at [developer.mend.io](https://developer.mend.io).
```

> 2. In the **Credentials** page, press `ADD SECRET` to add the plain text secret.

Reviewer 2: ```suggestion
2. On the **Credentials** page, select `ADD SECRET` to add the plaintext secret.
```

   ![Add repo secret](../assets/images/app-settings/add-repo-secret.png)

> 3. Provide a value for `Secret name`, put the secret in the `Secret Value` field, and press `SAVE`.

Reviewer 2: ```suggestion
3. Give a value for `Secret name`, paste the plaintext secret into the `Secret Value` field, and select `SAVE`.
```

   ![Add a Secret dialog box](../assets/images/app-settings/add-a-secret.png)

> 4. Look for a confirmation dialog showing **“Successfully stored secret”**.

Reviewer 2: ```suggestion
4. Wait for the confirmation dialog: **“Successfully stored secret”**.
```
